packages/cli/src/models/network/NetworkController.ts

```diff
@@ -410,24 +409,31 @@ export default class NetworkController {
   }
 
   // Contract model
-  public checkContractDeployed(packageName: string, contractAlias: string, throwIfFail = false): void {
+  public logErrorIfContractPackageIsInvalid(packageName: string, contractName: string, throwIfFail = false): void {
     if (!packageName) packageName = this.projectFile.name;
-    const err = this._errorForContractDeployed(packageName, contractAlias);
-    if (err) this._handleErrorMessage(err, throwIfFail);
+    const err = this.getPackageContractError(packageName, contractName);
+    if (err) this.logErrorMessage(err, throwIfFail);
+  }
+
+  // Contract model
+  public logErrorIfProjectDeploymentIsInvalid(throwIfFail = false): void {
+    const err = this.getDeploymentErrorForProject();
+    if (err) this.logErrorMessage(err, throwIfFail);
   }
 
   // Contract model
-  public checkLocalContractsDeployed(throwIfFail = false): void {
-    const err = this._errorForLocalContractsDeployed();
-    if (err) this._handleErrorMessage(err, throwIfFail);
+  public logErrorIfContractDeploymentIsInvalid(contractName: string, throwIfFail = false): void {
+    const err = this.getContractDeploymentError(contractName);
+    if (err) this.logErrorMessage(err, throwIfFail);
   }
 
   // Contract model
-  private _errorForLocalContractsDeployed(): string {
-    const [contractsDeployed, contractsMissing] = partition(this.projectFile.contractAliases, alias =>
-      this.isContractDeployed(alias),
+  private getDeploymentErrorForProject(): string {
+    const contractsMissing = this.projectFile.contracts.filter(
+      o => this.isProjectFileContract(o) && !this.isNetworkFileContract(o),
```

Comment:
I'm really curious why you're using the variable name `o`. :thinking:

Reply:
Personal preference. It is common to use `i` but I find `o` to be better suited because it is not take by loops and and can't be confused with `l`.